### ISSUE TITLE
error for bad backslashes gives unrelated error message with the wrong location

### ISSUE THREAD

**Issue by joachim-n** [Seldaek/jsonlint#54]
Given the following JSON:

```
{
  "SA-CORE-2019-002": "../patches/sa-core-2019-002.patch",
  "#3026386-26 - Drush fatal error after upgrading to 8.6.6, 8.5.9, or 7.62: PHP Fatal error: Uncaught TYPO3\PharStreamWrapper\Exception": "https://www.drupal.org/files/issues/2019-01-16/d8-3026386-26.patch"
}
```

the error is:

```
test.json: Parse error on line 2:
...-2019-002.patch",  "#3026386-26 - Drush
---------------------^
Invalid string, it appears you forgot to terminate a string, or attempted to write a multiline string which is invalid - It appears you have an extra trailing comma
```

The error message doesn't seem related, as there is no extra trailing comma.

**Comment by Seldaek** [comment]
I can confirm the issue but I am really not sure how to fix this unfortunately.

**Comment by stof** [comment]
the issue here is that the code figuring out this message relies on reading the upcoming input of the lexer to see the content of the broken string. But this upcoming input returns at most 20 chars, and the unescaped backslash is farther than that in the string. So it the parser does not detect the backslash.

**Comment by Seldaek** [comment]
Ah thanks @stof that was helpful to figure out the fix. It's not super elegant because in this state the lexer does not have a proper knowledge of the string yet, but I think it's a safe fix.
